collections/src/main/scala/strawman/collection/StrictOptimizedIterableOps.scala

```diff
@@ -53,6 +53,20 @@ trait StrictOptimizedIterableOps[+A, +CC[_], +C]
     (first.result(), second.result())
   }
 
+  override def unzip3[A1, A2, A3](implicit asTriple: A => (A1, A2, A3)): (CC[A1], CC[A2], CC[A3]) = {
+    val b1 = iterableFactory.newBuilder[A1]()
+    val b2 = iterableFactory.newBuilder[A2]()
+    val b3 = iterableFactory.newBuilder[A3]()
+
+    foreach { xyz =>
+      val (x, y, z) = asTriple(xyz)
```

Comment:
We should avoid destructuring assigments in performance-critical code: https://github.com/scala/bug/issues/10767